### DISCUSSION TITLE
How to get store values with composition api?

### DISCUSSION THREAD

**Discussion by whitespacecode** [CodeDredd/pinia-orm#337] [Q&A] [answered]
Using the composition api from vue it's not clear to me how you can access the values in the store and execute actions on the different items.
My setup:

```
const postRepo = useRepo(Post);
const posts = computed(() => postRepo.query().with('comments').get())

console.log(posts); //-> see log below
console.log(posts.value); //-> this is empty

//What i want to achieve 
onMounted(() => {
  posts.value.forEach(post => {
       //create something for each post
        console.log(post);
    });
});
```

Looking at the vue devtools the posts are there:
```
posts:Array[10] (Computed)
0:Object
1:Object
2:Object
3:Object
4:Object
5:Object
6:Object
7:Object
8:Object
9:Object
```

Log where you see the posts in the value 
```
ComputedRefImpl {dep: undefined, __v_isRef: true, _dirty: true, effect: ReactiveEffect, _setter: ƒ, …}
ReactiveEffect {active: true, deps: Array(181), parent: undefined, fn: ƒ, scheduler: ƒ, …}
_value: (10) [Post, Post, Post, Post, Post, Post, Post, Post, Post, Post]
value: Array(10)

```

**Comment by CodeDredd** [comment, marked as answer]
Can it be that your saving your data to the store from an api async and not waiting for it to be done?
Please make a reproduction on stackblitz or codesandbox 🙏 Thank you!

**Reply by whitespacecode** [reply]
i'm using Inertiajs. Indeed with using nextTick it 'solves' my problem.
I just want to do something after the posts are set.

Using the regular piniastore i could just 'use' and access the store with everything inside available
`const postStore = usePostStore();
`
```
postRepo.piniaStore().$onAction(({name}) => {
        if (name == "save") {
           nextTick(() => {
              const posts = postRepo.query().with('comments').get();
              console.log(posts);
            })
        }
    });
```

Now i can access my posts but i guess there must be a better way.

**Reply by CodeDredd** [reply]
Just use `Inertia.on('finish', () => ...)` to do the foreach. Thats pretty clean.
Look up https://inertiajs.com/progress-indicators
